### PR TITLE
add a document to help troubleshoot data transfers

### PR DESCRIPTION
**Pull request by whyrusleeping**
This is a very common scenario that people frequently ask us about on IRC. While we work on things that make it better, this guide should help anyone having issues.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: docs/file-transfer.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+external IP, then the network knows a valid external address for your node. At
+this point, its safe to assume that your node has a difficult to traverse NAT
+situation. If this is the case, you can try to enable upnp or NAT-PMP on the
+router of node A and retry the process. Otherwise, you can try manually
+connecting node A to node B.
+
+### Manually connecting node A to B On node B, run `ipfs id` and take one of
+the multiaddrs that contains its public ip address, and then on node A run
+`ipfs swarm connect <multiaddr>`.  If that *still* doesn't work, then you
+should either join IRC and ask for help there, or file an issue on github.
```
Comment (magik6k) [inline docs/file-transfer.md]: Some note on p2p-circuit may be helpful too

Comment (whyrusleeping) [reply docs/file-transfer.md]: @magik6k @vyzo mind suggesting some text for how to dial a circuit relay address here? (i'm not sure if you just `ipfs swarm connect /p2p-circuit/QmFooBar` or what)

Comment (magik6k) [reply docs/file-transfer.md]: I think it'd be enough to just link https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#circuit-relay

File: docs/file-transfer.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+## A file transfer
+To start, make sure that ipfs is running on both machines. To verify, run `ipfs
+id` on each machine and check if the `Addresses` field has anything in it. If
+it says `null`, then your node is not online and you will need to run `ipfs
+daemon`.
+
+Now, lets call the node with the file you want to transfer node 'A' and the
+node you want to get the file to node 'B'. On node A, add the file to ipfs
+using the `ipfs add` command. This will print out the multihash of the content
+you added. Now, on node B, you can fetch the content using `ipfs get <hash>`.
```
Comment (dignifiedquire) [inline docs/file-transfer.md]: I think it would be good to have an example here, of the output of the command

**Pull request merged**